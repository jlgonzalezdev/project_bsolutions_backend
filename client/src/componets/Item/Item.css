.Item{
    font-size: large;
    background-color: white;
    border-radius: 5px;
    padding: 5px;
    margin: 5px;
}

.hidden{
    display: none;
}

.itemDone{
    background-color: #83ff83;
}
