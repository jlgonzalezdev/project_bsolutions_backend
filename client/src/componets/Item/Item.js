import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import '@fortawesome/fontawesome-free/css/all.css';
import './Item.css';
export default class Item extends React.Component {

    constructor(props) {
        super(props);
        this.state = { isEditing: false, itemText: props.item.taskStr, itemTextOriginal: props.item.taskStr, isDone: props.item.isDone };
        this.toggleEdit = this.toggleEdit.bind(this);
        this.updateItem = this.updateItem.bind(this);
        this.cancelEdit = this.cancelEdit.bind(this);
        this.saveEdit = this.saveEdit.bind(this);
        this.toggleDone = this.toggleDone.bind(this);
    }

    toggleEdit(e) {
        this.setState({ isEditing: !this.state.isEditing });
    }

    toggleDone(e) {

        this.setState({ isDone: !this.state.isDone }, () => {
            this.props.updateItem(this.props.item, this.state.itemText, this.state.isDone);
        });
    }

    cancelEdit(e) {
        this.toggleEdit();
        this.setState({ itemText: this.state.itemTextOriginal });
    }

    saveEdit(e) {
        this.setState({ itemTextOriginal: this.state.itemText }, () => {
            this.props.updateItem(this.props.item, this.state.itemText, this.state.isDone);
        });
        this.toggleEdit();
    }

    updateItem(e) {
        this.setState({ itemText: e.target.value })
    }

    render() {
        return (<div className={this.state.isDone ? 'Item itemDone' : 'Item'}>
            <div className="row">
                <div className="col-8">
                    <div className={this.state.isEditing ? 'hidden' : ''} onClick={this.toggleEdit}>
                        <span className="fas fa-edit" style={{ padding: 5,fontSize:'small' }} ></span>
                        {this.state.itemText}
                    </div>
                    <div className={this.state.isEditing ? '' : 'hidden'}>
                        <input type="text" style={{ width: '100%' }} value={this.state.itemText} onChange={this.updateItem}></input>
                        <div className="row" style={{ padding: 20 }}>
                            <button style={{ margin: 5, fontSize: 'small' }} onClick={this.saveEdit}>Ok</button>
                            <button style={{ margin: 5, fontSize: 'small' }} onClick={this.cancelEdit}>Cancelar</button>
                        </div>
                    </div>
                </div>
                <div className="col-4">
                    <span className="fas fa-check" style={{ padding: 5 }} onClick={this.toggleDone}></span>
                    <span className="fas fa-times" style={{ padding: 5 }} onClick={() => { this.props.deleteItem(this.props.item) }}></span>
                </div>
            </div>
        </div>);
    }

}

