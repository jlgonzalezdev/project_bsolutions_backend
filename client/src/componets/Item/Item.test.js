import React from 'react';
import Item from './Item';
import renderer from 'react-test-renderer';
import Enzyme,{shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


Enzyme.configure({ adapter: new Adapter() });

describe('Testing UI', () => {
    test('has to set class itemDone when task is done', () => { 
        const component = renderer.create(
            <Item item={{ taskStr:'Task 01', isDone:true }}></Item>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('has to remove class itemDone when task is uncompleted', () => { 
        const component = renderer.create(
            <Item item={{ taskStr:'Task 01', isDone:false }}></Item>,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });    

    test('toggle  Edit Mode', () => { 
        const component = shallow(<Item item={{ taskStr:'Task 01', isDone:false }}></Item>);        
        component.find('#itemText').simulate('click');
        expect(component.state('isEditing')).toBe(true);

        component.find('#cancelBtn').simulate('click');
        expect(component.state('isEditing')).toBe(false);

    });    

    

});