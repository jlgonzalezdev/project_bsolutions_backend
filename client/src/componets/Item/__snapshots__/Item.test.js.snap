// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UI has to remove class itemDone when task is uncompleted 1`] = `
<div
  className="Item"
>
  <div
    className="row"
  >
    <div
      className="col-8"
    >
      <div
        className=""
        id="itemText"
        onClick={[Function]}
      >
        <span
          className="fas fa-edit"
          style={
            Object {
              "fontSize": "small",
              "padding": 5,
            }
          }
        />
        Task 01
      </div>
      <div
        className="hidden"
        id="itemTextEdit"
      >
        <input
          onChange={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
          type="text"
          value="Task 01"
        />
        <div
          className="row"
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <button
            id="okBtn"
            onClick={[Function]}
            style={
              Object {
                "fontSize": "small",
                "margin": 5,
              }
            }
          >
            Ok
          </button>
          <button
            id="cancelBtn"
            onClick={[Function]}
            style={
              Object {
                "fontSize": "small",
                "margin": 5,
              }
            }
          >
            Cancelar
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-4"
    >
      <span
        className="fas fa-check"
        onClick={[Function]}
        style={
          Object {
            "padding": 5,
          }
        }
      />
      <span
        className="fas fa-times"
        onClick={[Function]}
        style={
          Object {
            "padding": 5,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Testing UI has to set class itemDone when task is done 1`] = `
<div
  className="Item itemDone"
>
  <div
    className="row"
  >
    <div
      className="col-8"
    >
      <div
        className=""
        id="itemText"
        onClick={[Function]}
      >
        <span
          className="fas fa-edit"
          style={
            Object {
              "fontSize": "small",
              "padding": 5,
            }
          }
        />
        Task 01
      </div>
      <div
        className="hidden"
        id="itemTextEdit"
      >
        <input
          onChange={[Function]}
          style={
            Object {
              "width": "100%",
            }
          }
          type="text"
          value="Task 01"
        />
        <div
          className="row"
          style={
            Object {
              "padding": 20,
            }
          }
        >
          <button
            id="okBtn"
            onClick={[Function]}
            style={
              Object {
                "fontSize": "small",
                "margin": 5,
              }
            }
          >
            Ok
          </button>
          <button
            id="cancelBtn"
            onClick={[Function]}
            style={
              Object {
                "fontSize": "small",
                "margin": 5,
              }
            }
          >
            Cancelar
          </button>
        </div>
      </div>
    </div>
    <div
      className="col-4"
    >
      <span
        className="fas fa-check"
        onClick={[Function]}
        style={
          Object {
            "padding": 5,
          }
        }
      />
      <span
        className="fas fa-times"
        onClick={[Function]}
        style={
          Object {
            "padding": 5,
          }
        }
      />
    </div>
  </div>
</div>
`;
